resources/zone_id.js

```diff
@@ -0,0 +1,501 @@
+  'TheChrysalis': 426,
+  'TheChurningMists': 400,
+  'TheCopiedFactory': 882,
+  'TheCore': 857,
+  'TheCrownOfTheImmaculate': 846,
+  'TheCrownOfTheImmaculateExtreme': 848,
+  'TheCrystarium': 819,
+  'TheDancingPlague': 845,
+  'TheDancingPlagueExtreme': 858,
+  'TheDiademEasy': 512,
+  'TheDiademHard': 515,
+  'TheDiademHuntingGrounds': 625,
+  'TheDiademHuntingGroundsEasy': 624,
+  'TheDiademTrialsOfTheMatron': 656,
+  'TheDragonsNeck': 142,
+  'TheDravanianForelands': 398,
+  'TheDravanianHinterlands': 399,
+  'TheDrownedCityOfSkalla': 731,
+  'TheDungeonsOfLyheGhiah': 879,
+  'TheDuskVigil': 434,
+  'TheDyingGasp': 847,
+  'TheEighteenthFloor': 504,
+  'TheEmpty': 878,
+  'TheEpicOfAlexanderUltimate': 887,
+  'TheFaceOfTrueEvil': 709,
+  'TheFeastCustomMatchCrystalTower': 767,
+  'TheFeastCustomMatchFeastingGrounds': 619,
+  'TheFeastCustomMatchLichenweed': 646,
+  'TheFeastRanked': 765,
+  'TheFeastTeamRanked': 745,
+  'TheFeastTraining': 766,
+  'TheFieldsOfGloryShatter': 554,
+  'TheFinalCoilOfBahamutTurn1': 193,
+  'TheFinalCoilOfBahamutTurn2': 194,
+  'TheFinalCoilOfBahamutTurn3': 195,
+  'TheFinalCoilOfBahamutTurn4': 196,
+  'TheFinalStepsOfFaith': 559,
+  'TheFirmament': 886,
+  'TheFirstAltarOfDjananQhat': 683,
+  'TheForbiddenLandEurekaAnemos': 732,
+  'TheForbiddenLandEurekaHydatos': 827,
+  'TheForbiddenLandEurekaPagos': 763,
+  'TheForbiddenLandEurekaPyros': 795,
+  'TheFractalContinuum': 430,
+  'TheFractalContinuumHard': 743,
+  'TheFringes': 612,
+  'TheGhimlytDark': 793,
+  'TheGoblet': 341,
+  'TheGrandCosmos': 884,
+  'TheGreatGubalLibrary': 416,
+  'TheGreatGubalLibraryHard': 578,
+  'TheGreatHunt': 761,
+  'TheGreatHuntExtreme': 762,
+  'TheHardenedHeart': 873,
+  'TheHauntedManor': 571,
+  'TheHeartOfTheProblem': 718,
+  'TheHiddenCanalsOfUznair': 725,
+  'TheHourglass': 178,
+  'TheHouseOfTheFierce': 681,
+  'TheHowlingEye': 208,
+  'TheHowlingEyeExtreme': 297,
+  'TheHowlingEyeHard': 294,
+  'TheHuntersLegacy': 875,
+  'TheJadeStoa': 746,
+  'TheJadeStoaExtreme': 758,
+  'TheKeeperOfTheLake': 150,
+  'TheLabyrinthOfTheAncients': 174,
+  'TheLavenderBeds': 340,
+  'TheLightfeatherProvingGrounds': 439,
+  'TheLimitlessBlueExtreme': 447,
+  'TheLimitlessBlueHard': 436,
+  'TheLochs': 621,
+  'TheLostAndTheFound': 874,
+  'TheLostCanalsOfUznair': 712,
+  'TheLostCityOfAmdapor': 363,
+  'TheLostCityOfAmdaporHard': 519,
+  'TheMandervilleTables': 831,
+  'TheMinstrelsBalladHadessElegy': 885,
+  'TheMinstrelsBalladNidhoggsRage': 566,
+  'TheMinstrelsBalladShinryusDomain': 730,
+  'TheMinstrelsBalladThordansReign': 448,
+  'TheMinstrelsBalladTsukuyomisPain': 779,
+  'TheMinstrelsBalladUltimasBane': 348,
+  'TheMisery': 680,
+  'TheNavel': 206,
+  'TheNavelExtreme': 296,
+  'TheNavelHard': 293,
+  'TheOcular': 844,
+  'TheOrbonneMonastery': 826,
+  'TheOrphansAndTheBrokenBlade': 715,
+  'ThePalaceOfTheDead': 570,
+  'ThePalaceOfTheDeadFloors101_110': 598,
+  'ThePalaceOfTheDeadFloors111_120': 599,
```

Comment:
> Are those Heaven On High instances actually the ten sets of floors plus the roof, as it looks? If so, it might be worth supporting those just because combat can occur there?

Yes, given that the palace of the dead has unique names for each floor but Heaven on High does not, it seems likely all those collisions are the Heaven on High floors.  Do you have ids for these? Maybe this is a TODO followup sort of thing?

Reply:
No, I certainly don't. I was just asking because of the coincidence of 11 collisions. Definitely put a TODO in there. I know some of my static members might be going in there in the next week, I'll grab a log if they do.

Reply:
how do you get this list @quisquous ?
In ContentFinderConditions for example, they are all labeled correcty (e..g ID 540 would be Heaven-on-High (Floors 1-10))

Reply:
The ids here are TerritoryType.ID.  For example, uwu is: 

`777,"w1fz","ffxiv/wil_w1/fld/w1fz/level/w1fz",1,"???","","Ultimacy","w1fz/00","LoadingImage#1",2,17,539,False,26,False,0,True,False,False,False,,,,,"QuestBattle#0","",-1,292,-1,False,2,0,0,0,False,False,0,False,False`.

The `539` there refers to ContentFinderCondition.ID, e.g. 

`539,"w1fz","w1fz",1,"InstanceContent#30067",False,2,0,0,"Disciples of War or Magic","ContentMemberType#3",1,0,0,,0,70,70,0,375,False,False,False,False,False,0,False,True,True,False,False,False,True,False,False,False,"the Weapon's Refrain (Ultimate)","Ultimate Raids",0,,1002,"ui/icon/112000/112296.tex",,-1,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False`

One example of Heaven-on-High in TerritoryType is:

`770,"e3c1","ex2/02_est_e3/cnt/e3c1/level/e3c1",1,"Othard","Othard","Heaven-on-High","","LoadingImage#10",2,31,0,False,0,False,0,True,False,False,False,,,,,"QuestBattle#0","",-1,313,-1,False,2,0,0,0,False,False,0,False,False`

The value that is `539` from uwu is `0` in all of the HOH lines.  I don't know how to get from that line to `540` in ContentFinderCondition.

Reply:
TerritorryType 770 can be linkt via the "e3c1" to the same on ContentFinderCondition > TerritorryType column = "e3c1"

Reply:
Ok, using the territory name isn't perfect either, because they're not unique, but I can use that as a backup if the id isn't there.  That gives the following:

```diff
diff --git a/resources/zone_id.js b/resources/zone_id.js
index 53db82e2..d7a559ee 100644
--- a/resources/zone_id.js
+++ b/resources/zone_id.js
@@ -11,6 +11,7 @@ const ZoneId = {
   'ASleepDisturbed': 914,
   'ASpectacleForTheAges': 533,
   'AccrueEnmityFromMultipleTargets': 540,
+  'AirForceOne': 832,
   'AkadaemiaAnyder': 841,
   'AkhAfahAmphitheatreExtreme': 378,
   'AkhAfahAmphitheatreHard': 377,
@@ -77,6 +78,11 @@ const ZoneId = {
   'CentralAzysLla': 507,
   'CentralShroud': 148,
   'CentralThanalan': 141,
+  'ChocoboRaceCostaDelSol': 389,
+  'ChocoboRaceSagoliiRoad': 390,
+  'ChocoboRaceTranquilPaths': 391,
+  'ChocoboRaceTutorial': 417,
+  'ChocoboSquare': 388,
   'CinderDrift': 897,
   'CinderDriftExtreme': 912,
   'CloudNine': 429,
@@ -148,6 +154,7 @@ const ZoneId = {
   'FlickingSticksAndTakingNames': 219,
   'FortempsManor': 433,
   'Foundation': 418,
+  'FourPlayerMahjongKuitanDisabled': 831,
   'Gangos': 915,
   'Halatali': 162,
   'HalataliHard': 360,
@@ -159,6 +166,17 @@ const ZoneId = {
   'HealMultipleAllies': 550,
   'HeartOfTheCreator': 588,
   'HeartOfTheSworn': 210,
+  'HeavenOnHigh': 780,
+  'HeavenOnHighFloors11_20': 771,
+  'HeavenOnHighFloors1_10': 770,
+  'HeavenOnHighFloors21_30': 772,
+  'HeavenOnHighFloors31_40': 782,
+  'HeavenOnHighFloors41_50': 773,
+  'HeavenOnHighFloors51_60': 783,
+  'HeavenOnHighFloors61_70': 774,
+  'HeavenOnHighFloors71_80': 784,
+  'HeavenOnHigh': 780,
+  'HeavenOnHighFloors11_20': 771,
+  'HeavenOnHighFloors1_10': 770,
+  'HeavenOnHighFloors21_30': 772,
+  'HeavenOnHighFloors31_40': 782,
+  'HeavenOnHighFloors41_50': 773,
+  'HeavenOnHighFloors51_60': 783,
+  'HeavenOnHighFloors61_70': 774,
+  'HeavenOnHighFloors71_80': 784,
+  'HeavenOnHighFloors81_90': 775,
+  'HeavenOnHighFloors91_100': 785,
   'HellsKier': 810,
   'HellsKierExtreme': 811,
   'HellsLid': 742,
@@ -191,6 +209,7 @@ const ZoneId = {
   'LimsaLominsaUpperDecks': 128,
   'LongLiveTheQueen': 298,
   'LotusStand': 205,
+  'LovmMasterTournament': 506,
   'LovmPlayerBattleNonRp': 591,
   'LovmPlayerBattleRp': 589,
   'LovmTournament': 590,
@@ -315,6 +334,7 @@ const ZoneId = {
   'TheDiademHard': 515,
   'TheDiademHuntingGrounds': 625,
   'TheDiademHuntingGroundsEasy': 624,
+  'TheDiademTrialsOfTheFury': 630,
   'TheDiademTrialsOfTheMatron': 656,
   'TheDragonsNeck': 142,
   'TheDravanianForelands': 398,
@@ -350,6 +370,7 @@ const ZoneId = {
   'TheFringes': 612,
   'TheGhimlytDark': 793,
   'TheGoblet': 341,
+  'TheGoldSaucer': 144,
   'TheGrandCosmos': 884,
   'TheGreatGubalLibrary': 416,
   'TheGreatGubalLibraryHard': 578,
@@ -378,7 +399,6 @@ const ZoneId = {
   'TheLostCanalsOfUznair': 712,
   'TheLostCityOfAmdapor': 363,
   'TheLostCityOfAmdaporHard': 519,
-  'TheMandervilleTables': 831,
   'TheMinstrelsBalladHadessElegy': 885,
   'TheMinstrelsBalladNidhoggsRage': 566,
   'TheMinstrelsBalladShinryusDomain': 730,
```

There's still a couple dozen zones that have collisions, but I think that's probably ok? It'd be pretty easy to sort out the real ids for Uldah, New Gridania, Mor Dhona, etc if anybody wanted them.